qiita_db/test/test_data.py

```diff
@@ -478,6 +478,38 @@ def test_create_data_type_only(self):
         # preprocessed_data_id, filepath_id
         self.assertEqual(obs, [[3, obs_id - 1], [3, obs_id]])
 
+    def test_delete(self):
+        """Correctly deletes a preprocessed data"""
+        # testing regular delete
+        ppd = PreprocessedData.create(
+            self.study, self.params_table,
+            self.params_id, self.filepaths, prep_template=self.prep_template,
+            ebi_submission_accession=self.ebi_submission_accession,
+            ebi_study_accession=self.ebi_study_accession)
+        PreprocessedData.delete(ppd.id)
```

Comment:
Yes, I agree that it should be a separate issue. BTW, that concern is not solved with the current solution:

```python
PreprocessedData.delete(ppd.id)
ppd.whatever = 55
```

I moved the discussion here: https://github.com/biocore/qiita/issues/1045